pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,133 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
 
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+let timeout = null
+
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDiscoveryService = (defaultDataProviderUrl) => {
+  const [dataProviderUrl, setDataProviderUrl] = useState(defaultDataProviderUrl)
+  const [discoveredUrls, setDiscoveredUrls] = useState(null)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleDataProviderUrlChange = (newUrl) => {
+    setDiscoveredUrls(null)
+    setDataProviderUrl(newUrl)
+  }
+
+  useDebouncedEffect(() => {
+    if (dataProviderUrl === '') return
+
+    setIsLoading(true)
+
+    // TODO: Call API Here
+    clearTimeout(timeout)
+    timeout = setTimeout(() => {
+      if (dataProviderUrl === 'success.url')
+        setDiscoveredUrls([{ url: dataProviderUrl, isNew: true }])
+      else if (dataProviderUrl === 'duplicate.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }])
+      else if (dataProviderUrl === 'multiple-duplicates.url')
+        setDiscoveredUrls([{ url: dataProviderUrl }, { url: dataProviderUrl }])
+      else setDiscoveredUrls([])
+
+      setIsLoading(false)
+    }, 1000)
+  }, [dataProviderUrl])
+
+  useEffect(() => {
+    setDataProviderUrl(defaultDataProviderUrl)
+  }, [defaultDataProviderUrl])
+
+  return [
+    dataProviderUrl,
+    discoveredUrls,
+    isLoading,
+    handleDataProviderUrlChange,
+  ]
+}
+
+const getHelperMessage = (dataProviderUrl, discoveredUrls) => {
+  if (dataProviderUrl === '' || discoveredUrls === null) {
```

Comment:
There should be another type of message for `Invalid URL`. It should clearly say _The URL is invalid. Please, check it again_ if the URL is not valid. Since the validation is done on the browser side partially, we should either use ref + `checkValidity` API (preferable) or do it in the React (easier).

Reply:
Also, we should somehow identify if the URL seems to be OAI-PMH endpoint or any other URL.

Securely, it can be done using fetch with `?verb=Identify` and checking for 200/404 but we may face CORS issues. Some regular expression can also be applied but it gives us less liability.

I am imagining the case when we asked the user to provide exact OAI-PMH endpoint because we found a duplicate. The user does it but we got a duplicate again. We should say _It's added already_. Securely it can be done using (almost) exact matching of the OAI-PMH from CORE's duplicate and user's input.

Reply:
> we should either use ref + checkValidity API (preferable) or do it in the React (easier).

I was thinking about it a lot. `checkValidity` didn't suit for me because I wanted to avoid having `useDiscoveryService` hook clean and not dependent on some particular form also not to mention that `reportValidity()` shows message (for short period of time) in tooltip and it could be confusing for the user since we use helper message for it? If that's not the problem it can be changed. 


> Also, we should somehow identify if the URL seems to be OAI-PMH endpoint or any other URL.

This is something I would keep entirely on the backend. I don't think  frontend should fire any request.

Reply:
Yes, it should be like that but it's tricky. We should design the API to support this. In such case `POST /data-providers` should return an error message instead of the list. This should be discussed with the @oacore/api team.

Reply:
I still think that the message for the invalid URL should be _Invalid URL_ so we should do the validation somehow. If the `checkValidity` does not fit our needs, we should do it using a regexp on the client-side.

Doing it in JS will save us requests to the API.

Reply:
Please check it now. I used validation API to spare us lines of code.